documentation/config_file_puppetserver.md

```diff
@@ -27,24 +30,30 @@ The `puppetserver.conf` file contains settings for Puppet Server software. For a
+
+        For a value of `false`, Puppet uses the HOCON configuration file format and location. See the [`auth.conf` documentation](./config_file_auth.html) for more information.
 * The `profiler` settings configure profiling:
     * `enabled`: If this is set to `true`, Puppet Server enables profiling for the Puppet Ruby code. The default is `false`.
 * The `puppet-admin` section configures Puppet Server's administrative API. (This API is unavailable with Rack or WEBrick Puppet masters.) The settings in this section are now deprecated. Remove these settings and replace them with the authorization method that was introduced in Puppet Server 2.2, using a HOCON format for `auth.conf`. See the [`auth.conf` documentation][] for more information.
     * `authorization-required` determines whether a client certificate is required to access the endpoints in this API. If set to `false`, all requests will be permitted to access this API. If set to `true`, only the clients whose certnames are included in the `client-whitelist` setting are allowed access to the admin API. If this setting is not specified but the `client-whitelist` setting is specified, the default value for this setting is `true`.
     * `client-whitelist` contains an array of client certificate names that are allowed to access the admin API. Puppet Server denies any requests made to this endpoint that do not present a valid client certificate mentioned in this array.
 
-    If neither the `authorization-required` nor the `client-whitelist` settings are specified, Puppet Server uses the new authorization methods and [`auth.conf` documentation][] formats to access the admin API endpoints.
+        If neither the `authorization-required` nor the `client-whitelist` settings are specified, Puppet Server uses the new authorization methods and [`auth.conf` documentation][] formats to access the admin API endpoints.
+
+* The `versioned-code` settings configure commands required to use [static catalogs][]:
+    * `code-id-command` contains the path to an executable script that provides a `code_id`. When compiling a static catalog, Puppet Server uses the output of this script as the catalog's `code_id` to generate metadata for a specific version of the catalog's file resources. If `code-id-command` is not set and static catalogs are enabled, the catalog's `code_id` will be null.
+    * `code-content-command` contains the path to an executable script that Puppet Server invokes when an agent makes a [`static_file_content`][] API request for the contents of a [file resource][] with a `source` attribute. If `code-content-command` is not set and static catalogs are enabled, Puppet runs on a static catalog that attempt to fetch a file resource's content will fail.
```

Comment:
ah, right.  the difference here is that the test explicitly requests `/static_file_content`, whereas an agent will not do that, since there will be no `code_id`.